runtime/interpreter/value.go

```diff
@@ -15466,20 +15503,17 @@ func (v *CompositeValue) Transfer(
 	storable atree.Storable,
 ) Value {
 
-	baseUse, elementOverhead, dataUse, metaDataUse := common.NewCompositeMemoryUsages(v.dictionary.Count(), 0)
```

Comment:
For composites and dictionaries, `NewAtreeMapPreAllocatedElementsMemoryUsage` is used when creating a copy of a value.

Does this include / is this an alternative to the memory metered when a new empty value is created?

Said differently: Is it OK to remove these lines, or do they have to be moved above `NewMapFromBatchData`, or do these memory kinds not apply for it and `MemoryKindAtreeMapPreAllocatedElement` is used instead?

Reply:
@fxamacker I see. I tried to bring it back in [187ffc6](https://github.com/onflow/cadence/pull/2509/commits/187ffc6b45cb1c9f80dbeaf25758b0e16a163aac). Could you please have another look and see if that is correct?

Reply:
The update looks good! :+1: